src/app/app.component.ts

```diff
@@ -46,18 +47,31 @@ export class AppComponent implements OnInit {
             guesses: [...gameState.guesses, normalizedGuess],
           };
 
+          this.gameStateService.setGameState(updatedState);
+          
           if (normalizedGuess === gameState.landmarkName) {
             updatedState.status = 'correct';
+
+            const guessInput = document.querySelector('app-guess-input');
+            guessInput?.ariaDisabled;
+            
           } else if (updatedState.guesses.length >= 6) {
             updatedState.status = 'incorrect';
           }
 
-          this.gameStateService.setGameState(updatedState);
+          if (normalizedGuess !== gameState.landmarkName) {
+            this.gameStateService.revealHint();
+          }
```

Comment:
Could this be an else instead? 
```suggestion
          if (normalizedGuess === gameState.landmarkName) {
            updatedState.status = 'correct';
          } else if (updatedState.guesses.length >= 6) {
            updatedState.status = 'incorrect';
          } else {
            this.gameStateService.revealHint();
          }
```